freqtrade/templates/sample_hyperopt.py

```diff
@@ -172,3 +172,125 @@ def populate_sell_trend(dataframe: DataFrame, metadata: dict) -> DataFrame:
             return dataframe
 
         return populate_sell_trend
+
+    @staticmethod
+    def short_strategy_generator(params: Dict[str, Any]) -> Callable:
+        """
+        Define the short strategy parameters to be used by Hyperopt.
+        """
+        def populate_short_trend(dataframe: DataFrame, metadata: dict) -> DataFrame:
+            """
+            Buy strategy Hyperopt will build and use.
```

Comment:
As agreed and defined in #2183 - we will not have additional methods for this, but instead use "populate_buy_trend" / "populate_sell_trend" for this (potentially renamed to enter / exit).

Reply:
The indicator space for entering a long vs entering a short is going to be opposite

Reply:
Actually, I might have just been misunderstanding the hyperopt classes

Reply:
values can be different or identical - that's not really relevant (and fully depends on your strategy/hyperopt file).

Don't spend too much time with the hyperopt class itself (consider it legacy hyperopt method) - and instead look at StrategyParameters.

It's up to you there if you have a `buy_rsi` and `buy_rsi_short` parameter - or `buy_rsi` - and compare it `> 30` and `< 30` for long/short entries.

As said before (in the interface discussion) - the `populate_{buy, sell}_trend` is just a "thinking helper" - if you assign one column ('buy') - or both buy and sell in `populate_buy_trend` will not "really" matter (unless you overwrite it in the other method).